### DISCUSSION TITLE
Bagagem forte em Front-End ou ser FullStack?

### DISCUSSION THREAD

**Discussion by Marcelosilva10** [frontendbr/forum#135] [Discussão]
Fiz essa pergunta pela seguinte analise que fiz em alguns sites de emprego, quero ser um bom Front-end mais tenho observado ultimamente tem muitas empresas que querem pessoas que sejam boas em Front-End e também em Back-End praticamente um verdadeiro FullStack.

Por via de dúvidas vim pergunta para vocês qual e o mais ideal focalizar e uma tecnologia que seja praticamente só em Front end ou também saber várias tecnologias voltadas para área de Back-end?

**Comment by felipefialho** [comment]
Fala @Marcelosilva10, tomei a liberdade de alterar seu titulo, beleza?

**Comment by Marcelosilva10** [comment]
O sem problemas @LFeh obrigado!

**Comment by diessica** [comment]
Gente?

Essa pergunta é muito relativa. Só faz o que você mais gosta...

As empresas procuram perfis variados. Isso não tem que ser parâmetro.

**Comment by Marcelosilva10** [comment]
@diessica isso que é tenso, supondo você gosta muito de Front end, mais tem empresas que solicitam que você tenha uma forte bagagem também em Back-end, isso não pode se desqualificar para um processo seletivo pelo fato de não ter um qualidade boa em Back.

**Comment by baltazarparra** [comment]
Acho que uma das coisas mais fodas do front, é exatamente você ter N caminhos viáveis pra trilhar... Tem espaço pra tudo, o caminho só depende de você, não fique preso a meia dúzia de propostas, se especialize no que te dê prazer em trabalhar.

**Comment by danilovaz** [comment]
Acho que é bem como a @diessica citou: faz o que você gosta. Eu parto do princípio que conhecimento nunca é demais. Mas é interessante ter-se um foco no seu objetivo, assim você não dispersa tempo em estudos que não te agregam. É algo que vai de experimentar e se encontrar fazendo o que gosta. Quando tu faz algo que gosta, tu faz bem e dá resultados. Empresas querem resultados e nós, trabalhar com o que gostamos :heart:

**Comment by felipefialho** [comment]
Totalmente de acordo @diessica. 

Até porque ao meu ver, se tornar um Fullstack é algo que acontece naturalmente, a pessoa normalmente é mais ligada em alguma área especifica (Front ou Back) e com o passar dos anos acaba se tornando um Fullstack (caso deseje).

É sempre importante ter conhecimento sobre tudo que envolve sua profissão, mas não vejo uma necessidade de todos os profissionais serem Fullstack, varia de empresa para empresa, de pessoa para pessoa. 

Se você já tem isso como meta de carreira desde o inicio, corre sério risco de se tornar um Desenvolvedor Pato, ou seja: Sabe nadar, voar, andar, mas não faz nada direito. Isso porque especializações são muito importantes e é MUITO difícil se especializar totalmente em diversas áreas, tem um absurdo de coisas para aprender e não vão deixar que tenha um aprofundamento muito grande em nenhuma.

**Comment by Marcelosilva10** [comment]
@LFeh Infelizmente me tornei isso que falou dev pato, não sabe dev nada praticamente do zero, mais sabe um pouco de tudo, agora que estou mesmo focado em Front, mais seguindo conselhos de vocês acho que vou ter bons resultados daqui em diante.

**Comment by woliveiras** [comment]
Opa, estamos ae como exemplos de gente que perdeu o foco no começo por querer abraçar o mundo e virou pato! - Apesar do Pato ser um animal foda, o único que anda, voa e nada BEM, o que faz mal feito é o Ganso mano! kkkk

Eu também entrei sem ter um rumo (por isso que hoje fico tentando ajudar a galera a focar em alguma coisa via posts no Blog). Aprendi de tudo um pouco, por que, em alguns foruns, a galera indicava um monte de conhecimento pra quem quisesse "ser programador". Fora as vagas de trampo que pedem até que você saiba adestrar cachorro!

Só que fui aprendendo de tudo um pouco e não praticando quase nada, virei um patinho. E nos ultimos 2 anos (mais ou menos) que comecei a focar e vim pro Front de vez.

Confesso que isso me levou a ser bem frustrado no início de carreira, pois achava não saber nada e não conseguia performar no dia a dia. Foi a falta de **foco**.

Concordo com o @LFeh que se tornar o tal do Full Stack é um caminho natural pra gente, afinal de contas, uma hora você domina o que faz de tal maneira que sobra um tempinho pra aprender outras coisas e você vai pegando aos poucos (sem ficar louco. Porém tem muita gente que não curte isso e prefere não aprender outras áreas. É bom entender isso no seu perfil.

**Comment by felipefialho** [comment]
hahaha, não sabia essa do Ganso, até porque pra mim são praticamente o mesmo animal :smile:  

Legal seu depoimento @woliveiras. No meu perfil por exemplo, não me vejo me tornando um Full Stack ao longo do tempo, a tendencia é que vá me especializando cada vez mais em Front-End e melhorando minhas noções de UX. É o que eu gosto.

**Comment by felipefialho** [comment]
Só complementando para o @Marcelosilva10, rolou uma discussão bem legal entre as diversas sub-áreas de Front-End #32

**Comment by Marcelosilva10** [comment]
@LFeh opa vou ver aqui gostei de ver a indicações que vocês, como vocês mencionaram do pato que sabe tudo de um pouco e não sabe nada, trabalhei em um projeto aí que tinha toda a biblioteca pre definida já para o projeto então em questão de estrutura no HTML ficava tudo mais fácil, e assim você não consegue evoluir sabe, quando se torna uma pato parece que você estuda várias vezes uma coisa e não assimila por completo porque talvez você não leva aquilo como foco talvez.

**Comment by woliveiras** [comment]
[Nesse post](http://woliveiras.com.br/posts/um-ano-como-desenvolvedor-front-end/) contei um pouco de como foi meu primeiro ano como Front End. Só loucura!

**Comment by lagden** [comment]
fullstack é um título de reconhecimento...
é como ser lutador de MMA! você tem que ser faixa preta em uma arte marcial e depois aprender um pouco das outras...
ou seja, é só trabalhar bem que o resultado e reconhecimento vêm...

**Comment by doug2k1** [comment]
Eu já tentei ser bom em tudo, UX, back, front, devops, mas acabava me frustrando.

Hoje estou mais focado no frontend, mas não me arrependo de ter estudado outras coisas, e continuo estudando, até pelo prazer de aprender. Tento ser aquele profissional "T", que se aprofunda em um ponto (frontend, no meu caso), mas com conhecimento em outras áreas.

**Comment by raphaklaus** [comment]
@lagden muito boa a analogia com o MMA! Ahahah!

Acho que realmente o "título" de fullstack vem com o tempo, depois de você concluir certas trilhas no desenvolvimento de software. Em web por exemplo, eu comecei com front end, e depois por necessidade eu precisei ir pro back end... E hoje meus dois últimos projetos: um foi back end em NodeJS e outro front end com SVG e animações...

Não é obrigatório dominar os dois, você pode se especializar em um, ficar excelente, e no outro saber se virar bem. Mas o tempo automaticamente vai fazendo você melhorar as perícias que você não é bom...

**Comment by rubenmarcus** [comment]
Acho que antes de perguntar o que você quer ser, você tem que ser perguntar porque entrou na área.

Acho que todo mundo quer uma combinação de realização profissional + realização pessoal + realização financeira.. Acho que nessa conta, pra alguns tem mais valor certo aspecto e pra outros , outros aspectos. Por isso você ve tantos profissionais diferentes , uns focados em saber tudo que sai de novo, outros focados em conceitos, outros focados na comunidade, cada um tem um pouco mais puxado pra um lado.. 

Acho que a sua carreira, vai se moldar um pouco entre vontade + necessidades.. Nunca pode ser apenas um dos dois, porque você pode errar feio e ficar refem do mercado, ou refem do seu ego.. é bom sempre ir em palestras e eventos pra poder se avaliar e comparar com a realidade dos outros.

Acho que uma coisa que ninguem comenta, mas pra mim até os engineers, que são focados mais em código , tinham que ter o minimo de noção de design e UX...
